src/main/java/com/xjeffrose/chicago/client/RequestMuxer.java

```diff
@@ -116,56 +141,41 @@ private boolean blockAndAwaitPool() {
 
   public void write(T sendReq, SettableFuture<Boolean> f) {
     if (isRunning.get()) {
-      final MuxedMessage<T> mm = new MuxedMessage<>(sendReq, f);
-      messageQ.addLast(mm);
+      messageQ.addLast(new MuxedMessage<>(sendReq, f));
     }
   }
 
   private Channel requestNode() {
-    ChannelFuture cf = null;
-    try {
-      cf = connectionQ.poll(250, TimeUnit.MILLISECONDS);
-    } catch (InterruptedException e) {
-      rebuildConnectionQ(connectionQ);
-      blockAndAwaitPool();
-      try {
-        cf = connectionQ.poll(250, TimeUnit.MILLISECONDS);
-      } catch (InterruptedException e1) {
-        log.error("Could not establish viable connection to " + addr, e1);
-      }
-    }
+    ChannelFuture cf = connectionQ.pollFirst();
 
     if ((cf != null) && cf.isSuccess()) {
       if (cf.channel().isWritable()) {
+        connectionQ.addLast(cf);
         return cf.channel();
       } else {
-        connectionQ.remove(cf);
         rebuildConnectionQ(connectionQ);
         return requestNode();
       }
     } else {
-      connectionQ.remove(cf);
       rebuildConnectionQ(connectionQ);
       return requestNode();
     }
   }
 
   private void drainMessageQ() {
```

Comment:
consider renaming since it no longer drains the queue, just writes the first message if one exists.